pkg/roachprod/vm/gce/utils_test.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package gce
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/roachprod/config"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/vm"
+	"github.com/stretchr/testify/assert"
+)
+
+// TestWriteStartupScriptTemplate mainly tests the startup script tpl compiles.
+func TestWriteStartupScriptTemplate(t *testing.T) {
+	// writeStartupScript reads a public SSH key from the disk,
+	// so we need to mock the file to avoid a panic.
+	tempDir := t.TempDir()
+	err := os.WriteFile(fmt.Sprintf("%s/id_rsa", tempDir), []byte("dummy private key"), 0644)
+	assert.NoError(t, err)
+	err = os.WriteFile(fmt.Sprintf("%s/id_rsa.pub", tempDir), []byte("dummy public key"), 0644)
+	assert.NoError(t, err)
+
+	config.SSHDirectory = tempDir
+
+	// Actual test
+	_, err = writeStartupScript("", vm.Zfs, false, false, false)
```

Comment:
You could fairly easily convert this to an `echotest` which would let you visually inspect the script instead of just verifying it doesn't error out:

```
	// Actual test
	file, err := writeStartupScript("", vm.Zfs, false, false, false)
	assert.NoError(t, err)
	f, err := os.ReadFile(file)
	assert.NoError(t, err)
	echotest.Require(t, string(f), filepath.Join("testdata", "startup_script.sh"))
```

Although maybe people find these kinds of tests annoying as they are essentially change detection tests.

Reply:
Thanks for nudging me into doing it; I considered it, but had forgotten about `datadriven`, and it seemed like a lot of work for a change detection test without it.

With the `echotest`, it will make it much clearer when a startup script is updated and also what the startup scripts currently look like.